metadata-ingestion/src/datahub/ingestion/source_config/operation_config.py

```diff
@@ -15,6 +16,13 @@ class OperationConfig(ConfigModel):
         default=False,
         description="Whether to do profiling at lower freq or not. This does not do any scheduling just adds additional checks to when not to run profiling.",
     )
+
+    weekly_striping_enabled: bool = Field(
```

Comment:
Perhaps it was unclear in the notion doc, but we want to remove these "bespoke" config options in favor of a more general interface, where you can specify different operators to determine when profiling gets run.

This `OperationConfig` should be removed / repurposed to fit the config spec in the notion doc. That includes:
- Moving config around when to run profiling nested under `enabled`, which is currently a boolean
- Taking in a list of operators based on a name and list of values, rather than configs like `weekly_striping_enabled`, `profile_day_of_week`, `profile_date_of_month`
- Implementing each operator as a function, as specified in the notion doc
- Building an interface to execute each specified operator, which then gets called by each source when it's determining whether to run profiling. This could just be a method like `should_run_profiling` or something like that